Geometry/EcalCommonData/src/EcalPreshowerNumberingScheme.cc

```diff
@@ -106,7 +106,7 @@ uint32_t EcalPreshowerNumberingScheme::getUnitID(const EcalBaseNumber& baseNumbe
     }
 
     // Z index +Z = 1 ; -Z = 2
-    int zs = baseNumber.getCopyNumber("EREG");
+    int zs = baseNumber.getCopyNumber(5);
```

Comment:
```suggestion
    static constexpr int EREGHierarchyLevel = 5;
    int zs = baseNumber.getCopyNumber(EREGHierarchyLevel);
```

Reply:
It would be good to document where this number 5 comes from. My suggestion provides one possible name for the constant, but maybe you can think of a more accurate name.

Reply:
@cvuosalo yes the thing is that this function is already so full of magic numbers, that at this stage I give up ;p
To clean it, as mentioned I would actually not even use hierarchy levels and this function at all, but directly a hash map from G4 touchable sensitive volume to DetId, as is done for the Tracker. Since this full function actually re-computes for each ECAL SimHit an info we already have ;p

But on principle you are right, and I have made some minor renaming.